stake-pool/program/src/state.rs

```diff
@@ -534,6 +552,20 @@ pub struct Fee {
     pub numerator: u64,
 }
 
+impl Fee {
+    /// Applies the Fee's rates to a given amount, `amt`
+    /// returning the amount to be subtracted from it as fees
+    /// (0 if denominator is 0 or amt is 0),
+    /// or None if overflow occurs
+    pub fn apply(&self, amt: u128) -> Option<u128> {
+        if self.denominator == 0 {
+            return Some(0);
+        }
+        amt.checked_mul(self.numerator as u128)?
+            .checked_div(self.denominator as u128)
```

Comment:
Actually, this brings up a potential bug! If someone sets their fee quite low, there's a chance that someone can withdraw pico amounts to get this calculation to always be 0.  As such, this should return a minimum of 1 if numerator is > 0

Reply:
We were thinking that such errors were inconsequential as transaction fees would dominate at that point.

Reply:
Makes sense, all good then